Source/JavaScriptCore/runtime/ArrayPrototype.cpp

```diff
@@ -1513,7 +1513,23 @@ JSC_DEFINE_HOST_FUNCTION(arrayProtoFuncIndexOf, (JSGlobalObject* globalObject, C
     JSValue searchElement = callFrame->argument(0);
 
     if (LIKELY(isJSArray(thisObject))) {
-        JSValue result = fastIndexOf<IndexOfDirection::Forward>(globalObject, vm, asArray(thisObject), length, searchElement, index);
+        JSArray* array = asArray(thisObject);
+        Butterfly* butterfly = array->butterfly();
+        if (isCopyOnWrite(array->indexingMode()) && JSImmutableButterfly::isAtomStringsStructure(vm, butterfly) && searchElement.isString()) {
+            StringImpl* impl = asString(searchElement)->value(globalObject)->impl();
+            auto iter = vm.atomStringToJSStringMap.find(impl);
+            if (iter == vm.atomStringToJSStringMap.end())
+                return JSValue::encode(jsNumber(-1));
+
+            JSString* search = iter->value.get();
+            auto data = butterfly->contiguous().data();
+            for (unsigned i = 0; i < length; ++i) {
+                if (asString(data[i].get()) == search)
+                    return JSValue::encode(jsNumber(i));
+            }
```

Comment:
We can do same thing for `Array.prototype.includes`?

Reply:
Yeah, let's do that an other patch. Will add a FIXME.

Reply:
Added the FIXME here https://github.com/WebKit/WebKit/pull/41349/files#diff-6b24e225993b3ab3229b57e9279aae265badb1c9fd28f1a07d060073ecee8bf9R2156-R2158.